packages/dendron-next-server/pages/workspace/config.tsx

```diff
@@ -0,0 +1,326 @@
+import { DendronConfig } from "@dendronhq/common-all";
+import { Formik, Field, Form } from "formik";
+import React, { ReactNode, useState } from "react";
+import {
+  Box,
+  Button,
+  Code,
+  Container,
+  FormControl,
+  FormErrorMessage,
+  FormHelperText,
+  FormLabel,
+  Heading,
+  Input,
+  Stack,
+  Switch,
+  Text,
+  Tooltip,
+} from "@chakra-ui/react";
+import { get } from "lodash";
+
+const getConfigData = (): { data: DendronConfig } => {
+  return {
+    data: {
+      vaults: [
+        {
+          fsPath: "vault",
+          name: "",
+        },
+      ],
+      site: {
+        assetsPrefix: "",
+        copyAssets: false,
+        siteHierarchies: ["dendron"],
+        siteNotesDir: "",
+        siteRepoDir: "",
+        siteRootDir: "docs",
+        usePrettyRefs: false,
+      },
+    },
+  };
+};
+
+// const saveConfigData = (data: DendronConfig) => {
+//   console.log("data saved");
+//   console.log(JSON.stringify(data, null, 2));
+// };
+
+type InputControlProps = {
+  name: string;
+  label?: string;
+  placeholder: string;
+  help: string | ReactNode;
+  disabled?: boolean;
+};
```

Comment:
Should also take a `required` prop, as several fields shouldn't be blank.

Reply:
I'll add `readonly` fields as well to the `DendronConfig` file for fields that should not be manually edited

Reply:
`required` :heavy_check_mark: @ 464a9adf
(nothing to do with `readonly` field from `DendronConfig` yet)

Reply:
Note: when `readOnly` is present, we can utilize the `FormControl[isReadOnly]` prop.